app/src/main/java/io/gnosis/safe/ui/safe/add/AddSafeFragment.kt

```diff
@@ -8,27 +8,30 @@ import android.view.ViewGroup
 import androidx.lifecycle.Observer
 import androidx.navigation.fragment.findNavController
 import io.gnosis.safe.R
+import io.gnosis.safe.ScreenId
 import io.gnosis.safe.databinding.FragmentAddSafeBinding
 import io.gnosis.safe.di.components.ViewComponent
 import io.gnosis.safe.helpers.AddressInputHelper
-import io.gnosis.safe.ui.base.BaseFragment
 import io.gnosis.safe.ui.base.BaseStateViewModel
+import io.gnosis.safe.ui.base.BaseViewBindingFragment
 import kotlinx.android.synthetic.main.fragment_add_safe.*
 import pm.gnosis.model.Solidity
 import pm.gnosis.svalinn.common.utils.visible
 import pm.gnosis.utils.asEthereumAddressString
 import timber.log.Timber
 import javax.inject.Inject
 
-class AddSafeFragment : BaseFragment<FragmentAddSafeBinding>() {
+class AddSafeFragment : BaseViewBindingFragment<FragmentAddSafeBinding>() {
 
     @Inject
     lateinit var viewModel: AddSafeViewModel
 
     private val addressInputHelper by lazy {
-        AddressInputHelper(this, ::updateAddress, allowAddressBook = false)
+        AddressInputHelper(this, tracker, ::updateAddress, allowAddressBook = false)
```

Comment:
I suppose this shouldn't be necessary if we are going to use the singleton patterns with `Tracker`, you are still using DI here.

Reply:
Yes, injection is intended. Tracker is singleton, but we still use injection. This way it more convenient to unit test user property tracking